Makefile

```diff
@@ -1,199 +1,176 @@
+	@echo 'OPTIONS := CFG_NoN CFG_POINT_LIGHTING etc.'
+	@echo 'Then run `make your_custom_ucode_name`. See the Makefile for the list of options.'
+
+# With the defaults--all of these options unset--you get 2.08 (Banjo-Tooie),
+# which has none of the bugs / the latest versions of everything, except
+# G_BRANCH_Z (F3DEX2), no point lighting, and no NoN. Enabling the different
+# BUG_ options *gives* you the bugs--the default is always with the bugs fixed.
+# If you are modding and adding options, you just have to add them here, and
+# in the options list for your custom configuration.
+ALL_OPTIONS := \
+  CFG_G_BRANCH_W \
+  CFG_XBUS \
+  CFG_NoN \
+  CFG_POINT_LIGHTING \
+  CFG_OLD_TRI_WRITE \
+  CFG_EXTRA_0A_BEFORE_ID_STR \
+  CFG_G_SPECIAL_1_IS_RECALC_MVP \
+  CFG_CLIPPING_SUBDIVIDE_DESCENDING \
+  CFG_DONT_SKIP_FIRST_INSTR_NEW_UCODE \
+  BUG_CLIPPING_FAIL_WHEN_SUM_ZERO \
+  BUG_NO_CLAMP_SCREEN_Z_POSITIVE \
+  BUG_TEXGEN_LINEAR_CLOBBER_S_T \
+  BUG_WRONG_INIT_VZERO \
+  BUG_FAIL_IF_CARRY_SET_AT_INIT
+  
+ARMIPS ?= armips
+PARENT_OUTPUT_DIR ?= ./build
+ifeq ($(PARENT_OUTPUT_DIR),.)
+  $(error Cannot build directly in repo directory; see Makefile for details.)
+  # The problem is that we want to be able to have targets like F3DEX2_2.08,
+  # but this would also be the directory itself, whose existence and possible
+  # modification needs to be handled by the Makefile. It is possible to write
+  # the Makefile where the directory is the main target for that microcode, but
+  # this has worse behavior in case of modification to the directory. Worse, if
+  # it was done this way, then it would break if the user tried to set
+  # PARENT_OUTPUT_DIR anywhere else. So, better to support building everywhere
+  # but here than to support only building here.
+endif
+
+NO_COL := \033[0m
+RED    := \033[0;31m
+GREEN  := \033[0;32m
+YELLOW := \033[0;33m
+BLUE   := \033[0;34m
+INFO    := $(BLUE)
+SUCCESS := $(GREEN)
+FAILURE := $(RED)
+WARNING := $(YELLOW)
+
+$(PARENT_OUTPUT_DIR):
+	@printf "$(INFO)Creating parent output directory$(NO_COL)\n"
+ifeq ($(OS),Windows_NT)
+	mkdir $@
+else
+	mkdir -p $@
+endif
+ALL_UCODES :=
+ALL_UCODES_WITH_MD5S :=
+ALL_OUTPUT_DIRS :=
+
+define reset_vars
+  NAME := 
+  DESCRIPTION := (Not used in any retail games)
+  ID_STR := Custom F3DEX2-based microcode, github.com/Mr-Wiseguy/f3dex2 & Nintendo
+  ID_STR := RSP Gfx ucode F3DEX       fifo 2.04H Yoshitaka Yasumoto 1998 Nintendo.
+  MD5_CODE := 
+  MD5_DATA := 
+  OPTIONS := 
+  EXTRA_DEPS :=
+endef
+
+define ucode_rule
+  # Variables defined outside the function need one dollar sign, whereas
+  # variables defined within the function need two. This is because make first
+  # expands all this text, substituting single-dollar-sign variables, and then
+  # executes all of it, causing all the assignments to actually happen.
+  ifeq ($(NAME),)
+   $$(error Microcode name not set!)
+  endif
+  UCODE_OUTPUT_DIR := $(PARENT_OUTPUT_DIR)/$(NAME)
+  ifeq ($(OS),Windows_NT)
```

Comment:
This ifeq isn't needed anymore and breaks the build on my system when running under windows.